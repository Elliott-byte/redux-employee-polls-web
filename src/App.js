import { Fragment } from "react";
import Login from "./components/Login";
import { Routes, Route } from "react-router";
import Dashboard from "./components/Dashboard";
import Question from './components/Question';
import NewQuestion from "./components/NewQuestion";
import Leaderboard from "./components/Leaderboard";
import Navbar from "./components/Navbar";
import NotFound from "./components/NotFound";

function App() {
  return (<Fragment>
    <Navbar />
    <Routes>
      <Route path='/' exact element={<Login />} />
      <Route path='/dashboard' element={<Dashboard />} />
      <Route path='/question/:id' element={<Question />} />
      <Route path='/add' element={<NewQuestion />} />
      <Route path='/leaderboard' element={<Leaderboard />} />
      <Route path='*' element={<NotFound />} />
    </Routes>
  </Fragment>)
}

export default App;
