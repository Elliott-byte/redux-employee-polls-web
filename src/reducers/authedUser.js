import { SET_AUTHED_USER } from "../actions/authedUser";
import { LOGOUT_ACTION } from "../actions/loginAction";

export default function authedUser(state = null, action) {
	switch (action.type) {
		case SET_AUTHED_USER:
			return action.id;
		case LOGOUT_ACTION:
			return null;
		default:
			return state;
	}
}
