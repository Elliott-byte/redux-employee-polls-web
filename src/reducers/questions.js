import { ADD_QUESTION, ANSWER_QUESTION, RECEIVE_QUESTIONS } from "../actions/questions";

export default function questions(state = {}, action) {
	switch (action.type) {
		case RECEIVE_QUESTIONS:
			return {
				...state,
				...action.questions
			}
		case ANSWER_QUESTION:
			const { questionId, option, authedId } = action;

			return {
				...state,
				[questionId]: {
					...state[questionId],
					[option]: {
						...state[questionId][option],
						votes: state[questionId][option].votes.concat(authedId)
					}
				}
			};
		case ADD_QUESTION:
			return {
				...state,
				[action.question.id]: { ...action.question }
			}
		default:
			return state;
	}
}