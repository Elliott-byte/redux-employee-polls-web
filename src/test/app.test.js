import * as React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../reducers';

describe('Dashboard', () => {
	it('matches the snapshot', () => {
		const mockStore = createStore(reducers);
		const { asFragment } = render(
			<Provider store={mockStore}>
				<MemoryRouter>
					<App />
				</MemoryRouter>
			</Provider>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
