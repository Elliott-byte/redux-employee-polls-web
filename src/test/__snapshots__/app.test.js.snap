// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-gray-800"
    data-headlessui-state=""
  >
    <div
      class="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8"
    >
      <div
        class="relative flex h-16 items-center justify-between"
      >
        <div
          class="absolute inset-y-0 left-0 flex items-center sm:hidden"
        >
          <button
            aria-expanded="false"
            class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
            data-headlessui-state=""
            id="headlessui-disclosure-button-:r0:"
            type="button"
          >
            <span
              class="absolute -inset-0.5"
            />
            <span
              class="sr-only"
            >
              Open main menu
            </span>
            <svg
              aria-hidden="true"
              class="block h-6 w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start"
        >
          <div
            class="flex flex-shrink-0 items-center"
          >
            <img
              alt="Your Company"
              class="h-8 w-auto"
              src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=500"
            />
          </div>
          <div
            class="hidden sm:ml-6 sm:block"
          >
            <div
              class="flex space-x-4"
            >
              <a
                aria-current="page"
                class="bg-gray-900 text-white rounded-md px-3 py-2 text-sm font-medium"
                data-key="Home"
                href="/dashboard"
              >
                Home
              </a>
              <a
                class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                data-key="Leaderboard"
                href="/leaderboard"
              >
                Leaderboard
              </a>
              <a
                class="text-gray-300 hover:bg-gray-700 hover:text-white rounded-md px-3 py-2 text-sm font-medium"
                data-key="New"
                href="/add"
              >
                New
              </a>
            </div>
          </div>
        </div>
        <div
          class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
        >
          <div
            class="relative ml-3"
            data-headlessui-state=""
          >
            <div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="relative flex rounded-full bg-gray-800 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                data-headlessui-state=""
                id="headlessui-menu-button-:r1:"
                type="button"
              >
                <span
                  class="absolute -inset-1.5"
                />
                <span
                  class="sr-only"
                >
                  Open user menu
                </span>
                <img
                  alt=""
                  class="h-8 w-8 rounded-full"
                  src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8"
  >
    <div
      class="sm:mx-auto sm:w-full sm:max-w-sm"
    >
      <img
        alt="icon"
        class="mx-auto h-10 w-auto"
        src="/record-svgrepo-com.svg"
      />
      <h2
        class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
      >
        Employee Poll
      </h2>
    </div>
    <div
      class="flex items-center justify-center my-0 text-rose-500 h-8 mini-h-full"
    >
        
    </div>
    <div
      class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
    >
      <form
        action="#"
        class="space-y-6"
        method="POST"
      >
        <div>
          <label
            class="block text-sm font-medium leading-6 text-gray-900"
            for="email"
          >
            User
          </label>
          <div
            class="mt-2"
          >
            <input
              autocomplete="user"
              class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
              id="user"
              name="user"
              required=""
              type="user"
            />
          </div>
        </div>
        <div>
          <div
            class="flex items-center justify-between"
          >
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="password"
            >
              Password
            </label>
          </div>
          <div
            class="mt-2"
          >
            <input
              autocomplete="current-password"
              class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
              id="password"
              name="password"
              required=""
              type="password"
            />
          </div>
        </div>
        <div>
          <button
            class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
